code/modules/mob/living/burgerchat.dm

```diff
@@ -1,40 +1,93 @@
 // Thanks to Burger from Burgerstation for the foundation for this
 
-/* 
+/*
 BYOND Forum posts that helped me:
 http://www.byond.com/forum/post/1133166
 http://www.byond.com/forum/post/1072433
 http://www.byond.com/forum/post/940994
 http://www.byond.com/docs/ref/skinparams.html#Fonts
 */
 
+#define COLOR_JOB_UNKOWN "#dda583"
+#define COLOR_PERSON_UNKNOWN "#999999"
+
+GLOBAL_LIST_INIT(job_colors_pastel, list(
+	"Assistant" = 		"#bdbdbd",
+	"Captain" = 		"#FFDC9B",
+	"Head of Personnel" = "#4C4CDD",
+	"Bartender" = 		"#B2CEB3",
+	"Cook" = 			"#A2FBB9",
+	"Botanist" =		"#95DE85",
+	"Quartermaster" =	"#C79C52",
+	"Cargo Technician" ="#D3A372",
+	"Shaft Miner" =		"#CE957E",
+	"Clown" =			"#FF83D7",
+	"Mime" = 			"#BAD3BB",
+	"Janitor" = 		"#97FBEA",
+	"Curator" = 		"#A2FBB9",
+	"Lawyer" = 			"#C07D7D",
+	"Chaplain" =		"#8AB48C",
+	"Chief Engineer" = 	"#CFBB72",
+	"Station Engineer" ="#D9BC89",
+	"Atmospheric Technician" = "#D4A07D",
+	"Chief Medical Officer" = "#7A97DA",
+	"Medical Doctor" = 	"#6CB1C5",
+	"Chemist" = 		"#82BDCE",
+	"Geneticist" = 		"#83BBBF",
+	"Virologist" = 		"#75AEA3",
+	"Paramedic" = 		"#8FBEB4",
+	"Research Director"="#974EA9",
+	"Scientist" =		"#C772C7",
+	"Roboticist" = 		"#AC71BA",
+	"Head of Security" ="#D33049",
+	"Warden" = 			"#EA545E",
+	"Detective" = 		"#C78B8B",
+	"Security Officer" ="#E6A3A3",
+	"Brig Physician" = 	"#B364B3",
+	"Prisoner" = 		"#d38a5c",
+	"CentCom" = 		"#90FD6D",
+	"Unknown"=			COLOR_JOB_UNKOWN,
+))
+
 /mob/living
 	var/list/stored_chat_text = list()
 
+/proc/get_job_colour(job_title)
+	return GLOB.job_colors_pastel[job_title]
+
 /proc/animate_chat(mob/living/target, message, message_language, message_mode, list/show_to, duration)
 
 	var/static/list/chatOverhead_colors = list("#83c0dd","#8396dd","#9983dd","#dd83b6","#dd8383","#83dddc","#83dd9f","#a5dd83","#ddd983","#dda583","#dd8383")
-
-	var/text_color = pick(chatOverhead_colors)
+	var/text_color
+
+	var/mob/living/carbon/human/target_as_human = target
+	if(istype(target_as_human))
+		if(target_as_human.wear_id?.GetID())
+			var/datum/job/wearer_job = target_as_human.wear_id.GetJobName()
```

Comment:
This variable shouldn't work, but I am currently refactoring this to be easier to add new jobs